tests/exporter/collectors/test_cpu.py

```diff
@@ -36,18 +36,21 @@ def test_temp(self):
         else:
             self.assertIsNone(temp)
 
-    """
-    TODO: this test doesn't always pass.
+
+    # TODO: this test doesn't always pass.
     def test_str(self):
+        """
         @mccooeyc11
         Test whether the string representation of CPU metrics is correct
+        """
         cpu_collector = CPUCollector()
-        utilization = cpu_collector.get_utilization()
-        frequency = cpu_collector.get_frequency()
-        temp = cpu_collector.get_temperature().__str__()
-        expected_output = f"CPU utilization: {utilization}%\n CPU frequency: {frequency}MHz\n CPU temperature: {temp}°C\n"
-        self.assertEqual(str(cpu_collector), expected_output)
-    """
+        output_str = str(cpu_collector)
+        self.assertTrue('CPU utilization: ' in output_str)
+        self.assertTrue('%\n' in output_str)
+        self.assertTrue('CPU frequency: ' in output_str)
+        self.assertTrue('MHz\n' in output_str)
+        self.assertTrue('CPU temperature: ' in output_str)
```

Comment:
While this could be improved in the future to have a full regex expression to make sure all parts of the string are in the proper order, this definitely works and improves substantially on the previous version in terms of reliability.